### ISSUE TITLE
Manual mapping in "replace_ensembl_ids" function

### ISSUE THREAD

**Issue by dongbohu** [greenelab/generic-expression-patterns#14]
`replace_ensembl_ids()` function applies a `manual_mapping` dict:
https://github.com/greenelab/generic-expression-patterns/blob/23e67888e982bbbc3b0adf566604356e325a6bc5/generic_expression_patterns_modules/process.py#L57-L62
but it is applied before https://github.com/greenelab/generic-expression-patterns/blob/23e67888e982bbbc3b0adf566604356e325a6bc5/generic_expression_patterns_modules/process.py#L76-L78
so these four manual mappings don't seem to be included in the returned `gene_expression` dataframe, because the four new column names (`PLEKHG7`, `LINC00595`, `TAS2R45`, `CCL3L1`) are not indexed in `gene_id_mapping`.

**Comment by dongbohu** [comment]
The bug can be fixed like this (`manual_mapping` is passed in as an argument):
```python
def replace_ensembl_ids(original_expression, gene_id_mapping, manual_mapping):
    updated_mapping = gene_id_mapping.loc[~gene_id_mapping.index.duplicated(keep="first")]

    # Apply manual mapping here:
    for ensembl_id, gene_symbol in manual_mapping.items():
        updated_mapping.loc[ensembl_id].hgnc_symbol = gene_symbol

    updated_expression = original_expression.iloc[:, ~original_expression.columns.str.contains("PAR_Y")] 
    updated_expression.columns = updated_expression.columns.map(updated_mapping["hgnc_symbol"])

    updated_expression = updated_expression.iloc[:, updated_expression.columns != ""]
    updated_expression = updated_expression.iloc[:, updated_expression.columns.notnull()]

    return updated_expression
```

**Comment by ajlee21** [comment]
Thank you for finding this error. I will look into this. I'm not sure if switching the order as you suggested will necessarily fix things in this case. I think I might need to merge my `manual_mapping` and `gene_id_mapping` dictionaries and then apply them.

**Comment by dongbohu** [comment]
@ajlee21: The following two lines in my proposed code actually merged `manual_mapping` with `gene_id_mapping` (I am using another variable `updated_mapping` to avoid modifying the input parameter `gene_id_mapping` by accident):
```python
for ensembl_id, gene_symbol in manual_mapping.items():
        updated_mapping.loc[ensembl_id].hgnc_symbol = gene_symbol
```

**Comment by ajlee21** [comment]
I see, sorry, you're correct...I misread. Thank you!

**Comment by dongbohu** [comment]
Fixed in PR #19.
